flambe/nlp/classification/task.py

```diff
@@ -0,0 +1,204 @@
+    """A text classification task.
+
+    Performs text classifcation training and evaluation.
+    Takes as input a dataset and an encoder, and constructs
+    a simple TextClassifier. You may pass your custom fields
+    or used the defaults. The loss is computer via a cross entropy,
+    and the validation metric is accuracy.
+
+    """
+
+    def __init__(self,
+                 dataset: TabularDataset,
+                 encoder: Optional[Encoder] = None,
+                 pooling: Optional[Module] = None,
+                 dropout: float = 0,
+                 build_vocabularies: bool = True,
+                 train_batch_size: int = 32,
+                 val_batch_size: int = 32,
+                 embedding_args: Dict[str, Any] = None,
+                 optimizer: Optional[Optimizer] = None,
+                 iter_scheduler: Optional[LRScheduler] = None,
+                 eval_scheduler: Optional[LRScheduler] = None,
+                 text_field: Optional[TextField] = None,
+                 label_field: Optional[LabelField] = None,
+                 embedder: Optional[Embedder] = None,
+                 model: Optional[TextClassifier] = None,
+                 **kwargs) -> None:
+        """Initalize a TextClassification task.
+
+        Parameters
+        ----------
+        dataset : TabularDataset
+            The input dataset
+        encoder : Encoder
+            An encoder
+        pooling : Optional[Module], optional
+            A pooling method
+        dropout : float, optional
+            Dropout to apply between the encoder and output layer.
+            Default ``0``.
+        build_vocabularies : bool, optional
+            Whether the fields should expand their vocabulary Using
+            the training data. Default ``True``.
+        train_batch_size : int, optional
+            The batch size to use during. Default ``32``.
+        val_batch_size : int, optional
+            The batch size to use during evaluation. Default ``32``.
+        embedding_args : Dict[str, Any], optional
+            Keyword arguments to pass the ``Embeddings`` constructor.
+        optimizer : Optional[Optimizer], optional
+            The optimizer to use. Should be provided for training.
+        iter_scheduler : Optional[LRScheduler], optional
+            A learning rate scheduler to call on every training step.
+        eval_scheduler : Optional[LRScheduler], optional
+            A learning rate scheduler to call on every validation step.
+        text_field : Optional[TextField], optional
+            A custom text field to apply to the text inputs.
+        label_field : Optional[LabelField], optional
+            A custom label field to apply to the label inputs.
+        embedder : Optional[Embedder], optional
+            A custom embedder. Overrides ``encoder`` and ``pooling``.
+        model : Optional[TextClassifier], optional
+            A custom model. Overrides ``encoder`` and ``embedder``.
```

Comment:
I would provide factories to avoid confusion related to this. For example:

```
@staticmethod
def from_model():
   ...
```

Reply:
I think you mean classmethod, but yeah, we could do that. I don't think it's that much better though